src/classes/query.py

```diff
@@ -356,7 +356,7 @@ def filter(**kwargs):
                             object = Effect()
                             object.id = child["id"]
                             object.key = ["clips", {"id": clip["id"]}, "effects", {"id": object.id}]
-                            object.data = child
+                            object.data = copy.deepcopy(child)
```

Comment:
So, this was the ur-bug: unlike all of the other query classes which inherited their `filter()` method from `QueryObject`, the ad-hoc code in `Effect.filter()` failed to perform a `deepcopy()` of the relevant parts of the project data, instead returning a live reference that allowed the project data to be _directly_ modified. (All such modifications are supposed to go through the update manager.)